.github/workflows/code-styling.yml

```
@@ -15,9 +19,30 @@ concurrency:
 
 
 jobs:
+
+  setup-module-matrix:
+    runs-on: ubuntu-latest
+    name: Setup the module matrix
+    outputs:
+      matrix: ${{ steps.read-modules.outputs.matrix }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Read in list of mature analysis modules
+        id: read-modules
+        run: |
+          # From input `mature-modules.json`, read in modules which should be styled and save to $GITHUB_OUTPUT
+          echo "matrix=$(jq -cr . < ./.github/workflows/mature-modules.json)" > $GITHUB_OUTPUT
+
   style-code:
     runs-on: ubuntu-latest
     name: Perform code styling
+    needs: setup-module-matrix
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJSON(needs.setup-module-matrix.outputs.matrix) }}
```

I think for clarity it is probably worth making this a bit more explicit
```suggestion
      matrix: 
        module: ${{ fromJSON(needs.setup-module-matrix.outputs.matrix.modules) }}
```

Though see my overall comment about simplifying a bit so you can maybe just use `needs.setup-module-matrix.outputs.modules`